zaberMotionApp/src/zaberMotionRegister.cpp

```diff
@@ -0,0 +1,73 @@
+#include "zaberUtils.h"
+
+#include <epicsExport.h>
+#include <epicsStdio.h>
+#include <iocsh.h>
+
+#include "zaberController.h"
+#include "zaber/motion/library.h"
+
+// Zaber Motion Controller
+
+int zaberMotionCreateController(
+        const char *portName,   /* ZaberMotion Motor Asyn Port name */
+        int numAxes,            /* Number of axes this controller supports */
+        int movingPollPeriod,   /* Time to poll (msec) when an axis is in motion */
+        int idlePollPeriod,     /* Time to poll (msec) when an axis is idle. 0 for no polling */
+        const char *zaberPort,  /* Zaber Device TCP address or serial port name (prefixed with tcp:// or serial://) */
+        int zaberDeviceNumber   /* Zaber Device number on the port (1-indexed) */
+    )
+{
+    zaberController *pController = new zaberController(portName, numAxes, static_cast<double>(movingPollPeriod), static_cast<double>(idlePollPeriod), zaberPort, zaberDeviceNumber);
+    (void)pController;
+    return (asynSuccess);
```

Comment:
@kmpeters from what I've seen in the various motor repositories, it's common for these `CreateController` iocsh functions to return `asynSuccess` or `asynError`. However from what I can tell, this returned value is always ignored as the call func which is actually registered with the epics IOC returns void, and doesn't do any kind of status handling. It's the same in the newport motor repo for instance:
https://github.com/epics-motor/motorNewport/blob/16a818fe7e49746ffd1973c47354631cb0793e81/newportApp/src/XPSController.cpp#L1344

I'm wondering why these functions return anything at all. Are they called asynchronously anywhere?

Reply:
I think they return something because the return value could be checked, but in practice it never is.  I believe they are only called synchronously through the ioc shell.